apps/examples/mediarecorder/mediarecorder_main.cpp

```diff
@@ -234,28 +235,43 @@ class MediaRecorderTest : public MediaRecorderObserverInterface
 			return;
 		}
 
-		mp.setObserver(shared_from_this());
-		mp.setDataSource(std::move(source));
+		if (mp.setObserver(shared_from_this()) == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::setObserver failed" << std::endl;
+			
+			if (mp.destroy() == PLAYER_ERROR) {
```

Comment:
How about merging line 241, 250, and 259 by combining if logics in line 238, 247, 256?

Reply:
OK. 
I will merge this logic